recipes/hapcut2/meta.yaml

```diff
@@ -0,0 +1,42 @@
+package:
+  name: hapcut2
+  version: '1.1'
+
+source:
+  - url: https://github.com/vibansal/HapCUT2/archive/v1.1.tar.gz
+    sha256: f9fb1838b4c3f7fa4dc7bd6de72341f950caa3b2cf98b40a73927734dc75e26e
+    patches:
+    - shebang.patch
+    - flags.patch
+    - segfault.patch  # Patch from upstream master that’s not in the 1.1 release
+  - url: https://github.com/samtools/htslib/releases/download/1.2.1/htslib-1.2.1.tar.bz2
```

Comment:
Does it really require `htslib` and `samtools` as source in the `submodules`? Or would it be enough to add them in requirements? Both are in bioconda....

Reply:
I see.. it needs the includes of those two. Maybe its still enough to have them as dependency? What do you think @bioconda/review ?

Reply:
The Makefile wants to statically link against `libbam.a`, which is part of samtools. Version 1.2 of the samtools Bioconda package does not ship a `libbam.a`. I can find a `libbam.a` only in the ancient 0.1.19 version, but HapCUT2 wants version 1.2. None of the samtools recipes ship shared libraries, so using those is also not possible.

I don’t even know whether the shared or static libraries, such as `libbam.a`, that samtools creates when you build it are meant to be used from other projects (I thought the libraries were now in htslib), so I didn’t try to update the samtools recipe.

HapCUT2 also links against htslib (statically). I may be able to change it such that the Bioconda htslib is used instead, but then the recipe isn’t as "pure" as we want it anyway, so I decided not to do it. I can try if you prefer.

Reply:
And I forgot to mention: The samtools packages also don’t include `.h` files.

Reply:
Your sensible options are

1. Continue to bundle old samtools & htslib as per upstream and this recipe

2. Hope that HapCUT2 applies vibansal/HapCUT2#87 so it needs only htslib, and makes a new release, which you will be able to build against bioconda's htslib

3. Extract that PR as a patch to be applied by your recipe, and build against bioconda's htslib

Reply:
Cool, thanks so much! I think your PR has a good chance of getting merged considering what @pjedge wrote in vibansal/HapCUT2#86 . I’ll wait a bit and hope that we can go for option 2. I think any remaining cleanup can then wait for the next release.